build/xtask/src/config.rs

```diff
@@ -287,6 +288,81 @@ impl Config {
             })
             .collect()
     }
+
+    /// Calculates the output region which contains the given address
+    pub fn output_region(&self, vaddr: u64) -> Option<&str> {
+        let vaddr: u32 = match vaddr.try_into() {
+            Ok(v) => v,
+            Err(_) => return None,
+        };
+        self.outputs
+            .iter()
+            .find(|(_name, out)| {
+                vaddr >= out.address && vaddr < out.address + out.size
+            })
+            .map(|(name, _out)| name.as_str())
+    }
+
+    fn mpu_alignment(&self) -> MpuAlignment {
+        // ARMv6-M and ARMv7-M require that memory regions be a power of two.
+        // ARMv8-M does not.
+        match self.target.as_str() {
+            "thumbv8m.main-none-eabihf" => MpuAlignment::Chunk(32),
+            "thumbv7em-none-eabihf" | "thumbv6m-none-eabi" => {
+                MpuAlignment::PowerOfTwo
+            }
+            t => panic!("Unknown mpu requirements for target '{}'", t),
+        }
+    }
+
+    /// Checks whether the given chip's MPU requires power-of-two sized regions
+    pub fn mpu_power_of_two_required(&self) -> bool {
+        matches!(self.mpu_alignment(), MpuAlignment::PowerOfTwo)
```

Comment:
I feel like `self.mpu_alignment() == MpuAlignment::PowerOfTwo` is clearer, since you're not really using the Special Matches Powers

Reply:
Yeah, I was trying to avoid a `#[derive(..., PartialEq)]` block, but I suppose that's silly